web/tests/topic_generator.test.cjs

```diff
@@ -39,143 +36,144 @@ run_test("streams", ({override}) => {
     assert_prev_stream(1, 4);
 });
 
-run_test("topics", ({override}) => {
-    const streams = [1, 2, 3, 4];
-    const topics = new Map([
-        [1, ["read", "read", "1a", "1b", "read", "1c"]],
-        [2, []],
-        [3, ["3a", "read", "read", "3b", "read"]],
-        [4, ["4a"]],
-    ]);
-
-    function has_unread_messages(_stream, topic) {
-        return topic !== "read";
-    }
-
-    function get_topics(stream) {
-        return topics.get(stream);
-    }
-
-    function next_topic(curr_stream, curr_topic) {
-        return tg.next_topic(streams, get_topics, has_unread_messages, curr_stream, curr_topic);
-    }
-
-    assert.deepEqual(next_topic(1, "1a"), {stream_id: 1, topic: "1b"});
-    assert.deepEqual(next_topic(1, undefined), {stream_id: 1, topic: "1a"});
-    assert.deepEqual(next_topic(2, "bogus"), {stream_id: 3, topic: "3a"});
-    assert.deepEqual(next_topic(3, "3b"), {stream_id: 3, topic: "3a"});
-    assert.deepEqual(next_topic(4, "4a"), {stream_id: 1, topic: "1a"});
-    assert.deepEqual(next_topic(undefined, undefined), {stream_id: 1, topic: "1a"});
-
-    assert.deepEqual(
-        tg.next_topic(streams, get_topics, () => false, 1, "1a"),
-        undefined,
-    );
-
-    // Now test the deeper function that is wired up to
-    // real functions stream_data/stream_list_sort/unread.
-
-    const muted_stream_id = 400;
-    const devel_stream_id = 401;
-    const announce_stream_id = 402;
-    const test_here_stream_id = 403;
-    override(stream_list_sort, "get_stream_ids", () => [
-        announce_stream_id,
-        muted_stream_id,
-        devel_stream_id,
-        test_here_stream_id,
-    ]);
-
-    override(stream_topic_history, "get_recent_topic_names", (stream_id) => {
-        switch (stream_id) {
-            case muted_stream_id:
-                return ["ms-topic1", "ms-topic2", "unmuted", "followed-muted"];
-            case devel_stream_id:
-                return ["muted", "python", "followed-devel"];
-        }
-
-        return [];
-    });
-
-    override(stream_data, "is_muted", (stream_id) => stream_id === muted_stream_id);
```

Comment:
Hmm, I kinda feel like the right approach for this test is going to involve rewriting the test to just actually create the 7 topics that we're going to work with, rather than all these messy overrides? Not sure, I've not attempted it, and am not sure whether there's a simpler way to get the tests running... but this test in particular seems be like 50% overrides.